From bf73b2035c3d59baa7f484a84f7b5e60a9f58542 Mon Sep 17 00:00:00 2001
From: Beverly <beverlyt@google.com>
Date: Thu, 17 Apr 2025 16:24:22 +0000
Subject: [PATCH] Don't show dismissible keyguard in app pinning mode

On foldables, if the "Continue using apps on fold" is set
to "Swipe up to continue", when closing the device,
a dismissible keyguard will show.

If this setting is set, and there's a pinned app, instead
of showing the dismissible keyguard on fold, we keep the
screen on with the app displayed.
This keeps the app pinned and in the same security state as
when it was unfolded.

Bug: 404252173
Flag: EXEMPT bugfix
Test: manually (given settings described above), pin app, fold;
observe app still shows with screen on

Change-Id: I61e2014b9d21862859090707e5531a61bf966fc3
(cherry picked from commit 14f874decfdd5616b9fb3804154dd2560ebad0a1)
---
 .../com/android/systemui/keyguard/KeyguardService.java    | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardService.java b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardService.java
index d40fe468b0a5..13c72c67cdf4 100644
--- a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardService.java
+++ b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardService.java
@@ -331,6 +331,7 @@ public class KeyguardService extends Service {
         }
     };
     private final KeyguardServiceLockNowInteractor mKeyguardServiceLockNowInteractor;
+    private final ActivityManager mActivityManager;
 
     @Inject
     public KeyguardService(
@@ -350,6 +351,7 @@ public class KeyguardService extends Service {
             Lazy<SceneInteractor> sceneInteractorLazy,
             @Main Executor mainExecutor,
             KeyguardInteractor keyguardInteractor,
+	    ActivityManager activityManager,
             KeyguardEnabledInteractor keyguardEnabledInteractor,
             Lazy<KeyguardStateCallbackStartable> keyguardStateCallbackStartableLazy,
             KeyguardWakeDirectlyToGoneInteractor keyguardWakeDirectlyToGoneInteractor,
@@ -386,6 +388,7 @@ public class KeyguardService extends Service {
 
         mWmOcclusionManager = windowManagerOcclusionManager;
         mKeyguardEnabledInteractor = keyguardEnabledInteractor;
+	mActivityManager = activityManager;
         mKeyguardWakeDirectlyToGoneInteractor = keyguardWakeDirectlyToGoneInteractor;
         mKeyguardDismissInteractor = keyguardDismissInteractor;
         mKeyguardServiceLockNowInteractor = keyguardServiceLockNowInteractor;
@@ -677,6 +680,11 @@ public class KeyguardService extends Service {
             if (mFoldGracePeriodProvider.get().isEnabled()) {
                 mKeyguardInteractor.showDismissibleKeyguard();
             }
+
+            if (mActivityManager.getLockTaskModeState() != ActivityManager.LOCK_TASK_MODE_NONE) {
+                return;
+            }
+
             mKeyguardViewMediator.showDismissibleKeyguard();
 
             if (SceneContainerFlag.isEnabled() && mFoldGracePeriodProvider.get().isEnabled()) {
-- 
2.34.1

